celltype_annotation/analysis/copykat-rms-exploration.Rmd

```diff
@@ -0,0 +1,433 @@
+        output.seg = FALSE,
+        n.cores = params$threads
+      )
+      
+      # export full result
+      output_file <- file.path(output_dir,
+                              glue::glue("{params$library}_copykat_{distance_parameter}.rds"))
+      readr::write_rds(copykat_result, output_file)
+      
+            
+      # return predictions
+      return(copykat_result$prediction$copykat.pred)
+    }
+  )
+  
+  names(copykat_results) <- glue::glue("copykat_{distance_parameters}")
+  
+  # Bind up all the columns
+  copykat_df <- dplyr::bind_cols(
+    copykat_df, 
+    copykat_results
+  )
+  
+  # finally, export TSV with all copykat predictions
+  readr::write_tsv(copykat_df, copykat_distance_tsv)
+} 
+```
+
+### Comparing results
+
+
+We'll prepare the data for comparison and begin with a barplot of tumor/normal classifications across `copyKat` parameters:
+
+```{r}
+# Remove NA submitter values before assessment
+copykat_df <- copykat_df |>
+  tidyr::drop_na(submitter)
+
+# Plot the data:
+compare_copykat_results_barplot(copykat_df) + 
+  ggtitle("Comparing distance parameters")
+```
+
+Let's also look at the binary submitter celltypes to get a sense of balance:
+
+```{r}
+table(copykat_df$submitter_binary)
+```
+
+
+Next, we'll look at tables for correspondence between submitter cell type annotations and `copyKat` inferences, as well as confusion matrix results for each comparison.
+For `copyKat`, "diploid" is normal and "aneuploid" is tumor.
+
+```{r}
+copykat_df |>
+  dplyr::select(submitter, submitter_binary, copykat = copykat_euclidean) |>
+  compare_result_tables("euclidean distance") 
+
+copykat_df |>
+  dplyr::select(submitter, submitter_binary, copykat = copykat_pearson) |>
+  compare_result_tables("pearson distance") 
+
+copykat_df |>
+  dplyr::select(submitter, submitter_binary, copykat = copykat_spearman) |>
+  compare_result_tables("spearman distance") 
+
+```
+
+## Compare CopyKAT KS.cut parameters
+
+
+### Run copyKat
+
+In the previous section, we saw that correlational distance is likely to be better than euclidean.
+In this section, we'll again run `copyKat`, **but this time using _only_ the "pearson" distance**, at a variety of `KS.cut` parameters.
+
+As described in the [GitHub `README` file](https://github.com/navinlabcode/copykat/blob/b795ff793522499f814f6ae282aad1aab790902f/README.md?plain=1#L61), 
+
+> KS.cut is the segmentation parameter, ranging from 0 to 1. Increasing KS.cut decreases sensitivity, i.e. less segments/breakpoints. Usually it works in a range of 0.05-0.15.
+
+For this, we'll start with the SCE from the last section, and build it up from there with other `KS.cut` values.
+Since it was already run with default `KS.cut = 0.1`, we'll test a few additional values here and compare those to above.
+
+```{r}
+if (file.exists(copykat_kscut_tsv)) {
+  # if the TSV exists, read it in
+  copykat_df <- readr::read_tsv(copykat_kscut_tsv)
+
+} else if (file.exists(copykat_distance_tsv)) {
```

Comment:
I do think you could create a function that takes as input the tsv file and any parameters at the beginning of the notebook and then you won't have to rewrite this function again. Basically remove the `purrr` statement from here, make any parameters you want to test arguments with default values and then use `purrr` to run the function for the range of values for the parameter you are interested in. But I also don't know how much time we want to continue to spend on this.

Reply:
I thought about doing this actually when writing the notebook, but I didn't because it won't make much of a difference in terms of repeated code b/c each time we run `copyKat`, we vary a _different_ set of params. So, this still needs to have an `if` in this function to determine how to run (e.g. ,should it be `distance = the purrr variable to loop over` or `ks.cut = purrr variable to loop over`). 

That said, I agree it will clean up the notebook overall and make it easier to hone into relevant parts, so I'll go for it now!